src/corona_hakab_model/generation/matrix_generator.py

```diff
@@ -138,35 +122,23 @@ def __init__(
         self.connection_types_data: Dict[ConnectionTypes, ConnectionTypeData] = {
             con_type: matrix_consts.get_connection_type_data(con_type) for con_type in ConnectionTypes}
 
-        # TODO can we remove this? don't think we really support scipy matrices anymore
-        CoronaMatrix = corona_matrix.get_corona_matrix_class(matrix_consts.use_parasymbolic_matrix)
-        self.logger.info("Using CoronaMatrix of type {}".format(CoronaMatrix.__name__))
-        self.matrix = CoronaMatrix(self.size, self.depth)
-
         # create all sub matrices
-        with self.matrix.lock_rebuild():
-            # todo switch the depth logic, to get a connection type instead of int depth
-            current_depth = 0
-
-            for con_type, con_type_data in self.connection_types_data.items():
-                if con_type in Connect_To_All_types:
-                    self._create_fully_connected_circles_matrix(
-                        con_type_data, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                elif con_type in Random_Clustered_types:
-                    self._create_scale_free_graph(
-                        con_type_data, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                elif con_type in Geographic_Clustered_types:
-                    self._create_randomly_connected_layer(
-                        con_type_data, self.social_circles_by_connection_type[con_type], current_depth
-                    )
-                current_depth += 1
-
-        # current patch since matrix is un-serializable
-        self.matrix_data.matrix_type = "parasymbolic"
-        self.matrix_data.matrix = self.matrix
-        self.matrix_data.depth = self.depth
+        # todo switch the depth logic, to get a connection type instead of int depth
+        for current_depth, (con_type, con_type_data) in enumerate(self.connection_types_data.items()):
```

Comment:
Who you do not lock anymore?

Reply:
Cause you don't touch the matrix here, only the assignment array